### PR TITLE
Show EPG image on Dialog Channel Guide

### PR DESCRIPTION
**Pull request by lightglitch**
## Description
Show EPG program image on the guide dialog

## Screenshots (if appropriate):

image

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [x] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by ronie — CHANGES_REQUESTED** [review]
thanx for the pull-request!

there's currently a few issues though, when you focus the first item on the list, the image will be mostly offscreen.
also when the last item is focused, the plot will be partly offscreen.

perhaps it would be better to display the image on the left (or right) side of the plot instead of on top of it.

**Comment by lightglitch** [comment]
I'm not a designer so any suggestions to improve the interface are welcome.

**Comment by lightglitch** [comment]
Change position to the side:

image

**Review by ksooo — CHANGES_REQUESTED** [review]
I'm not a skinner, but tbh, this looks horrible. This definitely needs to visually reworked before it gets my approval.

**Comment by ksooo** [comment]
@ronie could we add this image to the list on the left, maybe right aligned, right to the show title?

**Comment by sualfred** [comment]
@ksooo 
I'm not quite sure, but isn't the channel logo used as icon if the EPG data has no thumb included?
Please keep in mind that there is no option to check if this ListItem.Icon is a EPG image or not. And a list full of the same images would look terrible.

Before something like this will be included it's necessary to store the EPG thumb in a own listitem to check it for existance (like @lightglitch tried in https://github.com/xbmc/xbmc/pull/14361 )

**Comment by ksooo** [comment]
@sualfred yes, this is all a big mess.

**Comment by sualfred** [comment]
Thanks for confirming. 

Anyway, an additional listitem for the EPG thumb is "overdue". From a skinner point of view it's annoying to have it available in some windows and in some not.  Correctly placed the thumb would fit very well in the OSD guide and other places like widgets. (OT and just my 2cts)

**Comment by lightglitch** [comment]
@sualfred That was my goal in the end. Because for example in the channel listing now it shows the channel icon on the left (small) and on the right (bigger) even when the epg has icon, with that change it's easy fixable.

**Comment by ksooo** [comment]
@lightglitch 

> That was my goal in the end.

 And I'm in no way against it. There is only one important constraint: It has to work, which was not the case with your PR. ;-) So, come up with a working and tested implementation and I will be happy to review it.

**Comment by lightglitch** [comment]
@ksooo I understand that's why I close it. Because I'm inexperienced in messing with xbmc core code if anyone can do it quickly fine by me.

**Comment by lightglitch** [comment]
So now that #14361 looks good what is the best approach for this one?
I also wanted to use the guiinfo label in the PVRInfoPanel (Includes_PVR.xml) as a replacement of the channel Icon if exists.

**Comment by lightglitch** [comment]
Like I said I'm not a designer but here is another option:
image
But I think the skinners of the project can find the best solution.

**Comment by ronie** [comment]
yup, that's what i had in mind when i asked to move the image to the left side of the plot.
looks good to me.

**Comment by lightglitch** [comment]
@ronie And if there's no image we move the plot to the left? Or redesign put the image on the right of the plot and just change the size of the background?

**Comment by ronie** [comment]
i would fall back to the channel icon if there's no epg image available

**Comment by lightglitch** [comment]
That is what is doing now but if you don't have channel icon stays an empty space.

**Comment by ronie** [comment]
if you don't have channel icons you'll end up with empty spaces in a lot of places elsewhere in the skin...
so i wouldn't consider that as a problem.

**Comment by ksooo** [comment]
Keeping aspect ratio is fine.

If you don't like the empty space if no channel logo is available you can add a fallback image to the texture tag. Please keep in mind that we have tv and radio channels.

At other places we use 
DefaultAudio.png for radio channels 
DefaultTVShows.png for TV channels

**Comment by ksooo** [comment]
Use String.IsEmpty(Window.Property(IsRadio)) to distinguish between tv and radio.

**Comment by ksooo** [comment]
You need two image controls, with different visible tags, then.

**Comment by lightglitch** [comment]
I'm updating the skin o use the EPGIcon Api, to simplify I was trying to use a $VAR but I doesn't work do I need some special configuration?

**Comment by ksooo** [comment]
I would stick with listitem.icon as channel logo as fallback is imo fine here. Plus as last resort, default images.

Not sure whether $VAR works for "visible" tags. You could look whether this is used like this elsewhere in the skin.

**Comment by lightglitch** [comment]
Update the pull with changes for all the section I could remember and test but it needs #14372 to work because of the change on the DialogSeekBar.xml section.

I also think the change in the home widget "WidgetListChannels" should use the $PARAM instead of the ListItem directly but I was unable to put it to work.
```
<visible>String.IsEmpty(ListItem.EpgEventIcon)</visible>
```

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: This make no sense as `Listitem.Icon` called on an epg event item has an automatic fallback to the channel icon.
A simple `<texture>$INFO[ListItem.Icon]</texture>` has the same effect.

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: No way to stroll in unrelated changes. Keep the changes to the topic of this PR, pleaes.

File: addons/skin.estuary/xml/DialogPVRChannelsOSD.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRChannelsOSD.xml]: makes no sense, see my other comment.

File: addons/skin.estuary/xml/DialogPVRChannelsOSD.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRChannelsOSD.xml]: Makes no sense, see my other comment.
Please revert all other changes of this kind you made.

File: addons/skin.estuary/xml/DialogPVRInfo.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRInfo.xml]: This makes zero sense as you state here that you want to show the epg event image only if its empty.

**Review by ksooo — CHANGES_REQUESTED** [review]
Sorry, please keep it simple.

File: addons/skin.estuary/xml/DialogSeekBar.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogSeekBar.xml]: Revert this. This has to be done in another way. As I wrote in the other PR, there must not be "VideoPlayer.EpgSomething" as VideoPlayer does not know about EPG. This is a PVR thing.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (lightglitch) [reply addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: ok, I will change so that if fallback is active don't change it.

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (lightglitch) [reply addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: ok

File: addons/skin.estuary/xml/DialogPVRChannelsOSD.xml
Comment (lightglitch) [reply addons/skin.estuary/xml/DialogPVRChannelsOSD.xml]: In this screen the Icon fallback is not active, we have to have it.

File: addons/skin.estuary/xml/DialogPVRChannelsOSD.xml
Comment (lightglitch) [reply addons/skin.estuary/xml/DialogPVRChannelsOSD.xml]: In this screen the Icon fallback is not active, we have to have it.

**Comment by ksooo** [comment]
>  lightglitch changed the title from Show EPG image on Dialog Channel Guide to Show EPG image on All PVR screens

 haha, no way, dude.

File: addons/skin.estuary/xml/DialogPVRChannelsOSD.xml
Comment (ksooo) [reply addons/skin.estuary/xml/DialogPVRChannelsOSD.xml]: Again, this is nonsense.

Comment (ksooo) [reply addons/skin.estuary/xml/DialogPVRChannelsOSD.xml]: Have you tested this or are you just guessing?

**Comment by lightglitch** [comment]
I'm still removing the changes you said to be out of scope, I'm just adding more screens to the pull request.

**Comment by ksooo** [comment]
@lightglitch Please, don't get me wrong, but I have the feeling that you present us with untested stuff here. This is a Nogo. I'm not your personal software tester. I will not spend any more of my free time on this.

**Comment by ksooo** [comment]
> I'm just adding more screens to the pull request.

 And I said to you that I don't want to have this in this PR.

**Comment by ksooo** [comment]
> make no sense just to change one screen at a time.

 Okay, let me state this a bit more clear. I'm the maintainer of the PVR component and I have a saying what make sense. This is how this project works.

**Comment by lightglitch** [comment]
@ksooo Sorry if you fell that way but I did tested on all the screen I made the changes. I can send you screenshots.

**Comment by lightglitch** [comment]
If you want me to revert everything I will do it without a problem.

**Comment by lightglitch** [comment]
@ksooo Done, reverted everything you asked.

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: This lines contains a bunch of trailing blanks. Could you please remove those.

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: If you change this line to `<texture fallback="DefaultTVShows.png">$INFO[Listitem.Icon]</texture>`you will get a default icon if neither an epg image nor a channel logo are available.

**Comment by ksooo** [comment]
Thanks, if you now could please squash down the commits to one, we are ready to merge.

**Comment by lightglitch** [comment]
Done.

On Aug 29, 2018 22:01, "Kai Sommerfeld" <notifications@github.com> wrote:

Thanks, if you now could please squash down the commits to one, we are
ready to merge.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/xbmc/xbmc/pull/14355#issuecomment-417104484>, or mute
the thread

**Comment by ksooo** [comment]
The commit message does not reflect the actual change you've done. You're not using any new API (aka gui labels) here. 

Sorry to be such a PITA, but I'm sure you now get the feeling that we also look at the smaller and surrounding things to keep up the quality of our software and can make it better with your next PRs, which are very welcome here.

**Comment by lightglitch** [comment]
Solved. No problem.

**Review by ksooo — APPROVED** [review]

**Review by ronie — APPROVED** [review]

**Pull request merged**
